lib/rubocop/cop/lint/no_untyped_raise.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # @example
+      #   # bad
+      #   raise 'foo'
+      #
+      #   # good
+      #   raise ArgumentError, 'foo'
+      #
+      class NoUntypedRaise < Cop
+        MSG = 'Do not raise untyped exceptions'.freeze
```

Comment:
I'd like to have a bit more explanation of why this is a bad thing, so it's more actionable / educational. e.g.:
```
Do not raise untyped exceptions, use a custom exception so it can be caught specifically.
```